stardata/Form1.cs

```diff
@@ -0,0 +1,235 @@
+                    string ABCshowLetters = "";
+                    string XYZoutput = "";
+                    string XYZshowLetters = "";
+
+                    string text = "Hr(RA) Min(RA) Sec(RA)  Deg(DEC) Hr(DEC) Sec(DEC)  Dst(PC) Rad  Nm";
+                   
+
+                    while ((line = file.ReadLine()) != null)
+                    {
+                        // Split the string into separate values
+                        string[] values = line.Split(' ');
+
+                        // Convert values into required data types
+
+                        int hoursRA = Math.Max(0, Math.Min(23, int.Parse(values[0])));
+                        int minutesRA = Math.Max(0, Math.Min(59, int.Parse(values[1])));
+                        double secondsRA = Math.Max(0, Math.Min(59.9999, double.Parse(values[2])));
+
+                        int degreesDEC = Math.Max(-360, Math.Min(360, int.Parse(values[3])));
+                        int minutesDEC = Math.Max(0, Math.Min(59, int.Parse(values[4])));
+                        double secondsDEC = Math.Max(0, Math.Min(59.9999, double.Parse(values[5])));
+
+                        double distance = Math.Max(0, Math.Min(40, double.Parse(values[6])));
+
+                        double Radius = 0;
+                        double.TryParse(values[7], out Radius);
+                        Radius = Math.Max(0, Math.Min(30, Radius));
+
+                        string Name = values[8];
+
+                        double A = Math.Round((hoursRA * 15) + (minutesRA * 0.25) + (secondsRA * 0.004166), 3);
+                        double B = Math.Round(Math.Abs(degreesDEC + (minutesDEC / 60) + (secondsDEC / 3600)) * Math.Sign(degreesDEC), 3);
+
+                        double X = (distance * Math.Cos(B)) * Math.Cos(A);
+                        double Y = (distance * Math.Cos(B)) * Math.Sin(A);
+                        double Z = distance * Math.Sin(B);
+
+                        // Add X, Y, Z values to lists
+                        xValues.Add(X);
+                        yValues.Add(Y);
+                        zValues.Add(Z);
+                        RadiusValues.Add(Radius);
+                        NameValues.Add(Name);
+
+
+                        // Format the output string into a table
+                        string lineOutput = $"{hoursRA,-9} {minutesRA,-12} {secondsRA,-7:F4} {degreesDEC,-14} {minutesDEC,-12} {secondsDEC,-10} {distance,-2} {Radius,-2} {Name,-0}\r\n";
+                        ABCshowLetters = $"{"A",-12} {"B",-12} {"C",0}";
+                        ABCoutput += $"\r{A,-8} {B,-8} {distance,-0}";
+
+                        XYZshowLetters = $"{"X",-32} {"Y",-32} {"Z",0}";
+                        XYZoutput += $"\r{X,-8} {Y,-8} {Z,-0}";
+
+
+
+
+                        // Add the string to the overall output
+                        output += lineOutput;
+                       
+                    }
+
+                    // Output the result to a text field

```

Comment:
consider dividing this method into to. one to read data and one to display data.